src/test/java/org/broadinstitute/consent/http/models/DatasetTests.java

```diff
@@ -18,4 +26,117 @@ public void testParseIdentifierToAlias() {
         assertThrows(IllegalArgumentException.class, ()->Dataset.parseIdentifierToAlias("DUOS-1234 56"));
         assertThrows(IllegalArgumentException.class, ()->Dataset.parseIdentifierToAlias("DUOS-1234as56"));
     }
+
+    @Test
+    public void testIsStringMatchName() {
+        String name = RandomStringUtils.randomAlphanumeric(20);
+
+        Dataset ds = new Dataset();
+        ds.setName(name);
+
+        assertTrue(ds.isStringMatch(name));
+        assertTrue(ds.isStringMatch(name.substring(5, 10)));
+        assertTrue(ds.isStringMatch(name.substring(10, 15)));
+
+        assertFalse(ds.isStringMatch(RandomStringUtils.randomAlphanumeric(30)));
+    }
+
+    @Test
+    public void testIsStringMatchNameCaseIndependent() {
+        String name = RandomStringUtils.randomAlphabetic(20);
+
+        Dataset ds = new Dataset();
+        ds.setName(name.toLowerCase());
+
+        assertTrue(ds.isStringMatch(name.toUpperCase()));
+        assertTrue(ds.isStringMatch(name.toUpperCase().substring(7, 14)));
+    }
+
+    @Test
+    public void testIsStringMatchDatasetProperty() {
+        Dataset ds = new Dataset();
+
+        String value = RandomStringUtils.randomAlphanumeric(20);
+
+        DatasetProperty dsp = new DatasetProperty();
+        dsp.setPropertyValue(value);
+        dsp.setPropertyType(DatasetPropertyType.String);
+        ds.setProperties(Set.of(dsp));
+
+        assertTrue(ds.isStringMatch(value));
+        assertFalse(ds.isStringMatch(RandomStringUtils.randomAlphanumeric(25)));
+    }
+
+    @Test
+    public void testIsStringMatchIdentifier() {
+        Dataset ds = new Dataset();
+        ds.setAlias(1235);
+
+        assertTrue(ds.isStringMatch("DUOS-001235"));
+        assertTrue(ds.isStringMatch("DUOS"));
+        assertTrue(ds.isStringMatch("123"));
+        assertTrue(ds.isStringMatch("001235"));
+        assertFalse(ds.isStringMatch("DUOS-123456"));
+    }
+
+    @Test
+    public void testIsStringMatchDataUseCommercial() {
+        Dataset ds = new Dataset();
+
+        assertFalse(ds.isStringMatch("collaborator"));
+
+        DataUse du = new DataUse();
+        du.setCollaboratorRequired(true);
```

Comment:
We have a handy builder object for this:
```suggestion
        DataUse du = new DataUseBuilder().setCollaboratorRequired(true).build();
```